flytekit/core/type_engine.py

```diff
@@ -353,7 +354,7 @@ def to_literal(self, ctx: FlyteContext, python_val: T, python_type: Type[T], exp
             )
         self._serialize_flyte_type(python_val, python_type)
         return Literal(
-            scalar=Scalar(generic=_json_format.Parse(cast(DataClassJsonMixin, python_val).to_json(), _struct.Struct()))
+            scalar=Scalar(generic=_json_format.Parse(cast(DataClassJSONMixin, python_val).to_json(), _struct.Struct()))
```

Comment:
Instead of using `cast`, we should apply mixins to a class instance after creation. check https://stackoverflow.com/questions/8544983/dynamically-mixin-a-base-class-to-an-instance-in-python.

`cast` does nothing at runtime.